ubuntu_application_sensors: Adapt to Android 7.1 API changes

fredldotme opened the pull request:
Use android::SensorManager::getInstanceForPackage() instead of android::SensorManager::getInstance() and set the event loop name.

APokorny commented:
I think you need to enclose the changes in ANDROID_VERSION_MAJOR >= 7 blocks.
Because we will probably build ubp-5.1 system images with that code.

fredldotme commented:
> I think you need to enclose the changes in ANDROID_VERSION_MAJOR >= 7 blocks.
> Because we will probably build ubp-5.1 system images with that code.

Agreed, should build now with 5.1 based device abstractions.

fredldotme commented:
Note: Build failures occur due to installing missing packages via apt

```
Err http://repo.ubports.com PR-3/main arm64 liburl-dispatcher1-dev arm64 0.2+ubports+0~20180611143914.3~1.gbp9a4659

  404  Not Found [IP: 2606:4700:20::6818:9 80]
```

Also, the changes are Android-side, which isn't handled by the build bot.

APokorny commented:
Yes you can ignore the build error. It is not related to the change. There seems to be something wrong with the o way the armhf build selects the repository fetch the url-dispatcher package from

APokorny commented on android/hybris/ubuntu_application_sensors_for_hybris.cpp:
```diff
@@ -77,9 +77,25 @@ static BackwardSensorTypeLut init_backward_sensor_type_lut()
     return lut;
 }
 
+#ifdef ANDROID_VERSION_MAJOR >= 7
```
#if ANDROID_VERSION_MAJOR >= 7

APokorny replied:
and same below ..

fredldotme replied:
alcohol

fredldotme committed:
ubuntu_application_sensors: Adapt to Android 7.1 API changes

Use android::SensorManager::getInstanceForPackage() and set the event loop name when building with Android headers >= 7,
Take the binary executable path as reference for the package and event loop name.
